src/metabase/query_processor/middleware/cache_backend/db.clj

```diff
@@ -92,15 +87,16 @@
 
 (defn- purge-old-cache-entries!
   "Delete any cache entries that are older than the global max age `max-cache-entry-age-seconds` (currently 3 months)."
-  [max-age-seconds]
-  {:pre [(number? max-age-seconds)]}
-  (log/tracef "Purging old cache entries.")
+  [criteria value]
+  (log/tracef "Purging cache entries.")
   (try
-    (t2/delete! (t2/table-name QueryCache)
-                :updated_at [:<= (seconds-ago max-age-seconds)])
+    (case criteria
+      :max-age-seconds (t2/delete! (t2/table-name QueryCache)
```

Comment:
how about a `:card-ids` criteria so we can do `DELETE query_cache where query_hash in (select cache_hash from query_execution where card_id in [....]` ?

I'm afraid there are too many hashes to put into an `in` condition. I haven't tested but it looks like there is a limit of 2k param for prepared statement: https://stackoverflow.com/questions/14631346/java-jdbc-prepared-statement-maximum-parameter-markers

Reply:
Yeah, forgot to split them in batches, but I'm going to do those by a thousand.

I can't pass card-ids because cache protocol has no idea about cards and if you ever implement like memcached then it won't make any sense.